From bd60a0c872416b752eb4242dfcc95abd15a7c85a Mon Sep 17 00:00:00 2001
From: bryango <bryango@users.noreply.github.com>
Date: Tue, 17 May 2022 20:33:11 +0800
Subject: [PATCH] Set ssl ciphers='ALL' in imaplib2 for Python 3.10

Fix SSLV3_ALERT_HANDSHAKE_FAILURE for some imap servers.
They use old, less secure ciphers that are disabled for Python 3.10.
See <https://stackoverflow.com/a/71007463>.
---
 Mailnag/common/imaplib2.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Mailnag/common/imaplib2.py b/Mailnag/common/imaplib2.py
index dfce0b4..0cbcff3 100644
--- a/Mailnag/common/imaplib2.py
+++ b/Mailnag/common/imaplib2.py
@@ -492,7 +492,7 @@ def ssl_wrap_socket(self):
 
             ssl_version =  TLS_MAP[self.tls_level][self.ssl_version]
 
-            self.sock = ssl.wrap_socket(self.sock, self.keyfile, self.certfile, ca_certs=self.ca_certs, cert_reqs=cert_reqs, ssl_version=ssl_version)
+            self.sock = ssl.wrap_socket(self.sock, self.keyfile, self.certfile, ca_certs=self.ca_certs, cert_reqs=cert_reqs, ssl_version=ssl_version, ciphers='ALL')
             ssl_exc = ssl.SSLError
             self.read_fd = self.sock.fileno()
         except ImportError:
